components/Panel/DelegationPanel.tsx

```diff
@@ -0,0 +1,231 @@
+import { Button, LoadingSpinner, PanelErrorBanner, TextInput } from "components";
+import { getAddress, isAddress } from "helpers";
+import { useDelegationContext, useErrorContext, usePanelContext, useUserContext } from "hooks";
+import NextLink from "next/link";
+import One from "public/assets/icons/one.svg";
+import Three from "public/assets/icons/three.svg";
+import Time from "public/assets/icons/time-with-inner-circle.svg";
+import Two from "public/assets/icons/two.svg";
+import { useEffect, useState } from "react";
+import styled, { css } from "styled-components";
+import { PanelFooter } from "./PanelFooter";
+import { PanelTitle } from "./PanelTitle";
+import { PanelSectionText, PanelSectionTitle, PanelWrapper } from "./styles";
+
+export function DelegationPanel() {
+  const { closePanel } = usePanelContext();
+  const { address } = useUserContext();
+  const { addErrorMessage, clearErrorMessages } = useErrorContext("delegation");
+  const [delegateAddressToAdd, setDelegateAddressToAdd] = useState("");
+  const {
+    getDelegationStatus,
+    sendRequestToBeDelegate,
+    getPendingSentRequestsToBeDelegate,
+    getDelegationDataFetching,
+  } = useDelegationContext();
+
+  const delegationStatus = getDelegationStatus();
+
+  useEffect(() => {
+    // only show this panel when the user has not yet entered a delegation relationship, or the user has requested another wallet to be their delegate wallet
+    if (!(delegationStatus === "no-delegation" || delegationStatus === "delegator-pending")) {
+      closePanel();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [delegationStatus]);
+
+  const pendingRequests = getPendingSentRequestsToBeDelegate();
+
+  // only allow user to add a delegate wallet if they are neither a delegator nor a delegate
+  const showAddDelegateInput = delegationStatus === "no-delegation";
+
+  const showPendingRequests = delegationStatus === "delegator-pending" && pendingRequests.length > 0;
```

Comment:
yeah you're right will add

Reply:
i think the only real benefit of having them defined in a constant is having a place to see all the options and document them, but yea i typically try to avoid defining them if i can. it just seems like this is used in a lot of places and theres like 5 different statuses to know. not super opinionated but i think it would help when someone else needs to debug or add feature

Reply:
I agree, it could be nice to define all the wallet status in an `enum WalletDelegationTypes{...}`

Some pros:
- readability: helps understand for example what a `"no-delegation"` is: it's a wallet delegation type or status
- maintainability: helps a new dev to understand the different types of delegation, modify them, fix them, etc... command clicking on the type and seeing the rest of types is helpful I think
- repetition: less repetition

Generally, I think that grouping elements/values of the same type of thing is useful. What do you think?

Reply:
The delegation status is defined by the type:

`
export type DelegationStatusT =
  | "no-wallet-connected"
  | "no-delegation"
  | "delegate"
  | "delegator"
  | "delegate-pending"
  | "delegator-pending";
  `
  
  The delegation status can only be one of these strings.
  
  To me this is:
  
  * more readable — `status === "no-wallet-connected"` vs `status === DelegationStatuses.NoWalletConnected`
  * more maintainable — no need to `import DelegationStatuses` and also editor autocomplete fills in the strings / takes you to the type definition
  * less repetition — with an enum you have to write `DelegationStatuses.` every time.
  
  Just my thoughts